### PR TITLE
Instant Search: Update styles to reduce usage of "!important".

### PR DESCRIPTION
**Pull request by a8ck3n**
#### Changes proposed in this Pull Request:

Introduces some more specific CSS rules for the various styled buttons in the Overlay. Separated out and limited to those styles (font/color attributes) that are currently dependant on `!important` directives. Necessary to prevent aggressive theme styles from bleeding into the Overlay.

With this technique, it would still be possible for site owners to override the styles as needed.

Also updates the font weights to match what we had initially before the Twenty Twenty One issues surfaced.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

1. Visit a site with a Jetpack Search plan.
2. Make sure the Twenty Twenty One theme is active. It has the aggressive theme styles we are trying to address.
3. Open the Overlay by appending `?s=&category=uncategorized` to your site address.
4. Confirm the Sort Options buttons look correct.
5. Confirm the "Clear filters" button looks correct.
6. Clear the filters and confirm the "Load more" button looks correct.
7. Inspect the various buttons and confirm there are no `!important` attributes used for size/color.

You can toggle the light/dark themes from `/wp-admin/admin.php?page=jetpack-search-configure`.

### CODE REVIEW

File: projects/packages/search/src/instant-search/lib/styles/_mixins.scss [hunk 1]
```diff
@@ -109,14 +109,14 @@ $customberg-container-selector: '.jp-search-configure-app-wrapper' !default;
 	letter-spacing: inherit;
 }
 
-// !important is temporarily needed to override :not() selectors in certain themes like Twenty Twenty One
-@mixin inherit-parent-colors-important() {
-	background-color: inherit !important;
-	color: inherit !important;
+@mixin apply-default-overlay-colors() {
+	background-color: inherit;
+	color: #646970;
```
Comment (kangzj) [inline projects/packages/search/src/instant-search/lib/styles/_mixins.scss:114]: The color look quite pale on dark theme,

image

Compared to trunk:

image

Could we use different colors for dark/light theme?

File: projects/packages/search/src/instant-search/components/search-filters.scss [hunk 2]
```diff
@@ -19,8 +19,6 @@
 }
 
 .jetpack-instant-search__clear-filters-link {
-	@include inherit-parent-colors-important();
-	@include inherit-parent-font-size-important();
 	border: none;
```
Comment (kangzj) [inline projects/packages/search/src/instant-search/components/search-filters.scss:22]: Is there any particular reason we didn't put these in the same selector on L36?

**Review by kangzj — COMMENTED** [review]
The overridden works as described. Left a couple of comments 😄

File: projects/packages/search/src/instant-search/components/search-filters.scss (hunk 2 above)
Comment (a8ck3n) [reply projects/packages/search/src/instant-search/components/search-filters.scss:22]: I'm not sure about that. Those changes were made while I was away so I wasn't part of any discussions that might have happened there. It was a bit of a rush to get the visual fixes into an Atomic update, I think.

That said, I could see an argument for keeping font metrics and colours as separate mixins.

File: projects/packages/search/src/instant-search/lib/styles/_mixins.scss (hunk 1 above)
Comment (a8ck3n) [reply projects/packages/search/src/instant-search/lib/styles/_mixins.scss:114]: Recent changes to fix the button styling did not include the font colors/weights we originally had. I didn't see any comment that this was intentional though, so this change reintroduces the values we've had all along. The white in your screenshot has only been a part of the styling for about 2 weeks. Prior to that, it looked like it does in this branch.

That said, we could definitely go with white (or close to it) but we'd be breaking from the predefined theme colors. I'll have another look at this and see what I can come up with.

Comment (a8ck3n) [reply projects/packages/search/src/instant-search/lib/styles/_mixins.scss:114]: After some more digging, I see we do have colours defined for these controls in our dark mode theme. They were just being overridden the same way the light colours were. I've added extra specificity to those rules so they should take effect correctly now. I'll add a screenshot to show the updated appearance.

**Comment by a8ck3n** [comment]
Hey @kangzj. Thanks for the review and comments!

I've got some changes in that enable the existing dark theme colours to work again. Per the original issue, this was a matter of overly aggressive rules in some themes overriding our styles. Some extra specificity in our existing dark theme CSS allows our colours to properly take effect.

SortOptions

**Comment by a8ck3n** [comment]
Merged changes from trunk, fixed up versions, and now passing CI tests. Ready for another review, please!

**Comment by oskosk** [comment]
This looks good.

Tested on JN using Twenty Twenty one... These seem to be the divergences between trunk and this branch. What's highlighted in green is what I see that differs from trunk

**Twenty Twenty one  running this branch**

image

**_Load more_**

image



**Twenty Twenty one  running Trunk**

image


**_Load more_**

image

**Review by oskosk — APPROVED** [review]

**Comment by a8ck3n** [comment]
Thanks for the review/merge @oskosk !

**Pull request merged**